### PR TITLE
Makefile.common: add 'update-go-deps' target

### PR DESCRIPTION
**Pull request by simonpasquier**
Also updated the RELEASE.md document to adjust the instructions about
dependencies management.

### CODE REVIEW

**Comment by roidelapluie** [comment]
should we call it update go-deps ?

**Comment by simonpasquier** [comment]
@roidelapluie sounds good, PR adjusted

**Comment by roidelapluie** [comment]
That does not remove existing indirects

File: Makefile.common [hunk 1]
```diff
@@ -150,6 +150,18 @@ else
 	$(GO) get $(GOOPTS) -t ./...
 endif
 
+.PHONY: update-go-deps
+update-go-deps:
+	@echo ">> updating Go dependencies"
+	@jq --version 1>/dev/null 2>&1 || { echo "Make sure that 'jq' is in your PATH"; exit 1; }
+	@for m in $$(go list -m -json all | jq -rc 'select(.Indirect != true and .Main != true) | .Path'); do \
```
Comment (roidelapluie) [inline Makefile.common:157]:
```suggestion
	@for m in $$(go list -m -f '{{ if and (not .Indirect) (not .Main)}}{{.Path}}{{end}}' all); do \
```

Comment (roidelapluie) [reply Makefile.common:157]: No need for jq

Comment (roidelapluie) [reply Makefile.common:157]: -mod=readonly is also needed

**Comment by simonpasquier** [comment]
> That does not remove existing indirects

I don't think it should in the scope of this change.

File: Makefile.common (hunk 1 above)
Comment (simonpasquier) [reply Makefile.common:157]: :+1:

File: Makefile.common [hunk 2]
```diff
@@ -150,6 +150,17 @@ else
 	$(GO) get $(GOOPTS) -t ./...
 endif
 
+.PHONY: update-go-deps
+update-go-deps:
+	@echo ">> updating Go dependencies"
+	@for m in $$(go list -mod=readonly -m -f '{{ if and (not .Indirect) (not .Main)}}{{.Path}}{{end}}' all); do \
```
Comment (roidelapluie) [inline Makefile.common:156]:
```suggestion
	@for m in $$($(GO) list -mod=readonly -m -f '{{ if and (not .Indirect) (not .Main)}}{{.Path}}{{end}}' all); do \
```

**Comment by roidelapluie** [comment]
:+1:

**Pull request merged**